### ISSUE TITLE
LZO corruption reported on intact files

### ISSUE THREAD

**Issue by xloem** [fox-it/dissect.squashfs#10]
```
import dissect.squashfs
import io, requests
file = io.BytesIO(requests.get('https://api.snapcraft.io/api/v1/snaps/download/XKEcBqPM06H1Z7zGOdG5fbICuf8NWK5R_2465.snap').content)
squashfs = dissect.squashfs.SquashFS(file)
inode = squashfs.get('/usr/share/groff/1.22.4/font/devlatin1/DESC')
print(inode.open().read())
```
```
  File "/home/user/.local/lib/python3.10/site-packages/dissect/squashfs/squashfs.py", line 434, in _read                                                                                    
    start_block, data = self.fs._read_block(start_block, run_block_size)                                                                                                                    
  File "/home/user/.local/lib/python3.10/site-packages/dissect/squashfs/squashfs.py", line 158, in _read_block                                                                              
    return block + length, self._compression.decompress(data, self.block_size)                                                                                                              
  File "/home/user/.local/lib/python3.10/site-packages/dissect/squashfs/compression.py", line 67, in decompress                                                                             
    return self._module.decompress(data, False, expected)                                                                                                                                   
lzo.error: Compressed data violation -6                                                                                                                                                     
```

**Comment by xloem** [comment]
I didn't end up investigating further myself, but when the other project addressed this and #11 in the linked issue and commits around that time, I think they said it involved LZO, extended inodes, and holes.

**Comment by Schamper** [comment]
This should be fixed with #12.
